File: 1399-count-largest-group/1399-count-largest-group.cs
public class Solution {
    public int CountLargestGroup(int n) {
        var maxSize = 0;
        var map = new Dictionary<int, int>();
        for(var num=1; num<=n; num++){
            var digitSum = GetDigitSum(num);
            if(!map.ContainsKey(digitSum)) map[digitSum] = 1;
            else map[digitSum] = 1 + map[digitSum];
            maxSize = Math.Max(maxSize, map[digitSum]);
        }

        var count = 0;
        foreach(var size in map.Values)
            if(size == maxSize) count++;

        return count;
    }

    private int GetDigitSum(int num){
        var digitSum = 0;
        while(num > 0){
            digitSum += num % 10;
            num /= 10;
        }
        return digitSum;
    }
}

File: 1399-count-largest-group/1399-count-largest-group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        max_size = 0
        count_map = defaultdict(int)

        for num in range(1, n + 1):
            digit_sum = self.get_digit_sum(num)
            count_map[digit_sum] += 1
            max_size = max(max_size, count_map[digit_sum])

        return sum(1 for size in count_map.values() if size == max_size)

    def get_digit_sum(self, num: int) -> int:
        digit_sum = 0
        while num > 0:
            digit_sum += num % 10
            num //= 10
        return digit_sum